python/tests/test_transforms.py

```diff
@@ -139,6 +143,104 @@ def test_string_with_surrogate_pair():
     assert bucket_transform(string_with_surrogate_pair) == mmh3.hash(as_bytes)
 
 
+@pytest.mark.parametrize(
+    "date,date_transform,expected",
+    [
+        (47, YearTransform(), "2017"),
+        (575, MonthTransform(), "2017-12"),
+        (17501, DayTransform(), "2017-12-01"),
+    ],
+)
+def test_date_to_human_string(date, date_transform, expected):
+    assert date_transform.to_human_string(DateType(), None) == "null"
+    assert date_transform.to_human_string(DateType(), date) == expected
+
+
+@pytest.mark.parametrize(
+    "timestamp,time_transform,expected",
+    [
+        (47, YearTransform(), "2017"),
+        (575, MonthTransform(), "2017-12"),
+        (17501, DayTransform(), "2017-12-01"),
+        (420042, HourTransform(), "2017-12-01-18"),
+    ],
+)
+def test_ts_to_human_string(timestamp, time_transform, expected):
+    assert time_transform.to_human_string(TimestampType(), None) == "null"
+    assert time_transform.to_human_string(TimestampType(), timestamp) == expected
+
+
+@pytest.mark.parametrize(
+    "type_var",
+    [
+        DateType(),
+        TimestampType(),
+        TimestamptzType(),
+    ],
+)
+def test_time_methods(type_var):
+    assert YearTransform().can_transform(type_var)
+    assert MonthTransform().can_transform(type_var)
+    assert DayTransform().can_transform(type_var)
+    assert YearTransform().preserves_order
+    assert MonthTransform().preserves_order
+    assert DayTransform().preserves_order
+    assert YearTransform().result_type(type_var) == IntegerType()
+    assert MonthTransform().result_type(type_var) == IntegerType()
+    assert DayTransform().result_type(type_var) == DateType()
+    assert YearTransform().dedup_name == "time"
+    assert MonthTransform().dedup_name == "time"
+    assert DayTransform().dedup_name == "time"
+
+
+@pytest.mark.parametrize(
+    "transform,type_var,value,expected",
+    [
+        (DayTransform(), DateType(), 17501, 17501),
+        (MonthTransform(), DateType(), 17501, 575),
+        (YearTransform(), DateType(), 17501, 47),
+        (YearTransform(), TimestampType(), 1512151975038194, 47),
+        (MonthTransform(), TimestamptzType(), 1512151975038194, 575),
+        (DayTransform(), TimestampType(), 1512151975038194, 17501),
```

Comment:
Can you add an hour test here as well?

Reply:
The hour transform has been tested separately in `test_hour_method`.